src/game.h

```diff
@@ -1083,6 +1085,10 @@ class game
         // Input related
         // Handles box showing items under mouse
         bool handle_mouseview( input_context &ctxt, std::string &action );
+        // See input_context::mouse_edge_scrolling(); this is a
+        // variant suitable for the terrain view since it takes the
+        // zoom level into account.
+        tripoint mouse_edge_scrolling_terrain( input_context ctxt );
```

Comment:
This makes a copy of `input_context`. Apart from inefficiency, there can be a bug caused by this.

Reply:
Cheers for the excellent review, @codemime! I'll wait for your input before I start resolving any of the above issues.